src/toolkits/drawing_classifier/drawing_classifier.cpp

```diff
@@ -661,6 +661,23 @@ gl_sframe drawing_classifier::perform_inference(data_iterator* data) const {
   return writer.close();
 }
 
+gl_sframe infer_feature_column_and_prepare_data(gl_sframe &data) {
+  std::string feature_column_name;
+
+  for (std::string column_name: data.column_names()) {
+    if (data[column_name].dtype() == flex_type_enum::LIST
+      || data[column_name].dtype() == flex_type_enum::IMAGE) {
+      feature_column_name = column_name;
```

Comment:
What happens if there's more than one matching column? (What do other toolkits do in similar scenarios?)

Reply:
Good point. Following what we were doing earlier in https://github.com/apple/turicreate/blob/master/src/python/turicreate/toolkits/_internal_utils.py#L122, looks like we error out in all scenarios except when there is exactly one candidate feature column.

However, looking at the `predict` docstring, I see:
```
            If dataset is an SFrame, it must have a column with the same name
            as the feature column during training. Additional columns are
            ignored.
```

So should we save the `feature_column_name_` as a private/public variable during training and just make sure the data passed in for testing has that same column? It does bring into question what our strategy would be if the user passed in an SArray. In order to make a `predict` call from the Python layer, we would want to convert the SArray to an SFrame at the Python layer and hence, know the feature column name at the Python layer. Alternatively, we could have a check in the C++ layer for a one-column SFrame in which case we rename the SFrame before continuing further processing.

What are your thoughts @nickjong ?

Reply:
Let's have the final result match the behavior we had in the original implementation. It's pretty typical to expect the prediction SFrame to use the same column name as the training data.

Reply:
Fixed.